frontend/src/features/admin-form/common/components/PreviewFormBanner/PreviewFormBanner.tsx

```diff
@@ -1,39 +1,197 @@
+                <Link
+                  variant="standalone"
+                  aria-label="Click to return to the admin dashboard"
+                  as={ReactLink}
+                  to={DASHBOARD_ROUTE}
+                >
+                  Back to FormSG
+                </Link>
+                <Button
+                  aria-label="Click to use this template"
+                  onClick={onModalOpen}
+                >
+                  Use this template
+                </Button>
+              </Stack>
+              <IconButton
+                color="primary.500"
+                variant="clear"
+                display={{ base: 'flex', md: 'none' }}
+                aria-label="Template preview actions"
+                onClick={onDrawerOpen}
+                icon={<BiDotsHorizontalRounded />}
+              />
+            </>
+          ) : (
+            <Button
+              aria-label="Click to edit the form"
+              as={ReactLink}
+              to={`${ADMINFORM_ROUTE}/${formId}`}
+            >
+              Edit form
+            </Button>
+          )}
+        </Flex>
+        <DuplicateFormModal
+          formId={formId}
+          isTemplate
+          isOpen={isModalOpen}
+          onClose={onModalClose}
+        />
+        <Drawer
+          placement="bottom"
+          onClose={onDrawerClose}
+          isOpen={isDrawerOpen}
+        >
+          <DrawerOverlay />
+          <DrawerContent borderTopRadius="0.25rem">
+            <DrawerBody px={0} py="0.5rem">
+              <Button
+                onClick={onModalOpen}
+                isFullWidth={true}
+                {...mobileDrawerButtonProps}
+              >
+                Use this template
+              </Button>
+              <Divider />
+              <Button
+                as={ReactLink}
+                to={DASHBOARD_ROUTE}
+                leftIcon={<BiArrowBack fontSize="1.25rem" />}
+                {...mobileDrawerButtonProps}
+              >
+                Back to FormSG
+              </Button>
+            </DrawerBody>
+          </DrawerContent>
+        </Drawer>
       </Flex>
-    </Flex>
+      <Divider />
+    </>
+  )
+}
+
+export const PreviewFormBannerContainer = ({
+  isTemplate,
+}: PreviewFormBannerProps): JSX.Element => {
+  const { isOpen, onOpen, onClose } = useDisclosure()
+  const handlePositionChange = useCallback(
+    (pos: Waypoint.CallbackArgs) => {
+      // Required so a page that loads in the middle of the page can still
+      // trigger the mini header.
+      if (pos.currentPosition === 'above') {
+        onOpen()
+      } else {
+        onClose()
+      }
+    },
+    [onClose, onOpen],
+  )
+
+  return (
+    <>
+      {isTemplate ? <StickyPreviewHeader isOpen={isOpen} /> : null}
+      <PreviewFormBanner isTemplate={isTemplate} />
+      {
+        /* Sentinel to know when sticky navbar is starting */
+
+        <Waypoint topOffset="64px" onPositionChange={handlePositionChange} />
```

Comment:
I see we already have a Sentinel waypoint in the mini header. Do that and this preview banner ever show together? If yes do the waypoints work well together?

Reply:
yup, the sticky use-template preview banner and the mini header show up in use-template mode. they seem to work well together! the waypoints are specific to each element